src/main/java/com/ibm/watson/modelmesh/ModelMesh.java

```diff
@@ -5907,10 +5927,22 @@ public void run() {
                                 if (ce == null) {
                                     ce = runtimeCache.getQuietly(modelId);
                                 }
-                                long lastUsed = -1L;
+                                long lastUsed = -2L;
                                 boolean remLoaded = loaded && (ce == null || ce.isFailed());
-                                boolean remFailed = failedTime != null
-                                        && ((ce != null && !ce.isFailed()) || now - failedTime > LOAD_FAILURE_EXPIRY_MS);
+                                boolean remFailed = false;
+                                if (failedTime != null) {
+                                    if (ce != null && !ce.isFailed()) {
+                                        remFailed = true;
+                                    } else {
+                                        lastUsed = ce != null ? runtimeCache.getLastUsedTime(modelId) : -1L;
+                                        // Use shorter expiry age if model was used in last 3 minutes
+                                        final long expiryAge = (lastUsed > 0 && (now - lastUsed) < 180_000L)
```

Comment:
should the _last 3 minutes_ (`180_000L`) here be a defined constant/env variable?

Reply:
Yes it could be ... let me push a change with that